metadata/metadata.go

```diff
@@ -47,6 +50,25 @@ func (i Info) Validate() error {
 	return nil
 }
 
+func decode(p []byte, data Validater) error {
+	dec := json.NewDecoder(bytes.NewReader(p))
+	for {
+		if err := dec.Decode(data); err != io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (i *Info) Write(p []byte) (n int, err error) {
```

Comment:
Looks like when you are trying to implement `io.Writer` interface go insist on using some special function signatures. Must be as is.